otcextensions/sdk/modelartsv1/v1/builtin_model.py

```diff
@@ -0,0 +1,162 @@
+    required = resource.Body("required", type=bool)
+    #: Parameter value.
+    value = resource.Body("value")
+
+
+class BuiltInModel(resource.Resource):
+    base_path = "/built-in-algorithms"
+
+    resources_key = "models"
+
+    # capabilities
+    allow_list = True
+    allow_fetch = True
+
+    _query_mapping = resource.QueryParameters(
+        "limit",
+        "order",
+        "offset",
+        "search_content",
+        "sort_by",
+        limit="per_page",
+        offset="page",
+        sort_by="sortBy",
+    )
+
+    # Properties
+    #: Time when a model is created.
+    created_at = resource.Body("create_time", type=int)
+    #: Engine ID of a model.
+    engine_id = resource.Body("engine_id", type=int)
+    #: Engine name of a model.
+    engine_name = resource.Body("engine_name")
+    #: Engine version of a model.
+    engine_version = resource.Body("engine_version")
+    #: Whether the request is successful.
+    is_success = resource.Body("is_success", type=bool)
+    #: Dataset format required by a model.
+    model_dataset_format = resource.Body("model_dataset_format")
+    #: URL of the model description.
+    model_description_url = resource.Body("model_description_url")
+    #: Model ID.
+    model_id = resource.Body("model_id")
+    #: Model name.
+    model_name = resource.Body("model_name")
+    #: Model precision.
+    model_precision = resource.Body("model_precision")
+    #: Model size, in bytes.
+    model_size = resource.Body("model_size", type=int)
+    #: Model training dataset.
+    model_train_dataset = resource.Body("model_train_dataset")
+    #: Model usage.
+    model_usage = resource.Body("model_usage", type=int)
+    #: Model name.
+    # name = resource.Body("name", alias="model_name")
+    #: Running parameters of a model.
+    parameter = resource.Body("parameter", type=list, list_type=ParameterSpec)
+
+    @classmethod
+    def existing(cls, connection=None, **kwargs):
+        """Create an instance of an existing remote resource.
+
+        When creating the instance set the ``_synchronized`` parameter
+        of :class:`Resource` to ``True`` to indicate that it represents the
+        state of an existing server-side resource. As such, all attributes
+        passed in ``**kwargs`` are considered "clean", such that an immediate
+        :meth:`update` call would not generate a body of attributes to be
+        modified on the server.
+
+        :param dict kwargs: Each of the named arguments will be set as
+            attributes on the resulting Resource object.
+        """
+        parameter = kwargs.get("parameter")
+        if parameter and isinstance(parameter, str):
+            kwargs["parameter"] = json.loads(parameter)
+        return cls(_synchronized=True, connection=connection, **kwargs)
+
+    @classmethod
+    def _get_one_match(cls, name_or_id, results):
+        """Given a list of results, return the match"""
+        the_result = None
+        for maybe_result in results:
```

Comment:
better to write specifically through which data we iterate and what is in the result

Reply:
this is a method of resource.Resource class , here its overridden since BuiltInModel class doesnt contains name and id attrs